testing/element-v1.js

```diff
@@ -0,0 +1,225 @@
+      const hasCallback =
+        implClass.prototype.onLayoutMeasure !==
+        BaseElement.prototype.onLayoutMeasure;
+      return !hasCallback;
+    },
+  },
+  {
+    name: 'Must not use getLayoutBox',
+    test: (implClass) => {
+      return !sourceIncludes(implClass, 'getLayoutBox');
+    },
+  },
+  {
+    name: 'Must not use getLayoutSize',
+    test: (implClass) => {
+      return !sourceIncludes(implClass, 'getLayoutSize');
+    },
+  },
+
+  {
+    name: 'Must not have pauseCallback',
+    test: (implClass) => {
+      const hasCallback =
+        implClass.prototype.pauseCallback !==
+        BaseElement.prototype.pauseCallback;
+      return !hasCallback;
+    },
+  },
+  {
+    name: 'Must not have resumeCallback',
+    test: (implClass) => {
+      const hasCallback =
+        implClass.prototype.resumeCallback !==
+        BaseElement.prototype.resumeCallback;
+      return !hasCallback;
+    },
+  },
+  {
+    name: 'Must not have unlayoutCallback',
+    test: (implClass) => {
+      const hasCallback =
+        implClass.prototype.unlayoutCallback !==
+        BaseElement.prototype.unlayoutCallback;
+      return !hasCallback;
+    },
+  },
+  {
+    name: 'Must not have unlayoutOnPause',
+    test: (implClass) => {
+      const hasCallback =
+        implClass.prototype.unlayoutOnPause !==
+        BaseElement.prototype.unlayoutOnPause;
+      return !hasCallback;
+    },
+  },
+  {
+    name: 'Must not have firstLayoutCompleted',
+    test: (implClass) => {
+      const hasCallback =
+        implClass.prototype.firstLayoutCompleted !==
+        BaseElement.prototype.firstLayoutCompleted;
+      return !hasCallback;
+    },
+  },
+  {
+    name: 'Must not have renderOutsideViewport',
+    test: (implClass) => {
+      const hasCallback =
+        implClass.prototype.renderOutsideViewport !==
+        BaseElement.prototype.renderOutsideViewport;
+      return !hasCallback;
+    },
+  },
+  {
+    name: 'Must not have idleRenderOutsideViewport',
+    test: (implClass) => {
+      const hasCallback =
+        implClass.prototype.idleRenderOutsideViewport !==
+        BaseElement.prototype.idleRenderOutsideViewport;
+      return !hasCallback;
+    },
+  },
+  {
+    name: 'Must not have isRelayoutNeeded',
+    test: (implClass) => {
+      const hasCallback =
+        implClass.prototype.isRelayoutNeeded !==
+        BaseElement.prototype.isRelayoutNeeded;
+      return !hasCallback;
+    },
+  },
+];
+
+/**
+ * @param {typeof BaseElement} implClass
+ * @param {{
+ *   exceptions: (!Array<string>|undefined),
+ * }=} options
+ */
+export function testElementV1(implClass, options = {}) {
+  const exceptions = options.exceptions || [];
+  RULES.forEach(({name, test}) => {
+    if (exceptions.includes(name)) {
+      expect(test(implClass), 'unused exception: ' + name).to.be.false;
+    } else {
+      expect(test(implClass), name).to.be.true;
+    }
+  });
+}
+
+function sourceIncludes(implClass, substring) {
```

Comment:
Nit: Can you add a fn comment explaining what this returns.

Reply:
Added jsdocs with description